### DISCUSSION TITLE
Issue with hidden field and multi definition list?

### DISCUSSION THREAD

**Discussion by eonpatapon** [cuelang/cue#417] [General] [answered]
```
#A: {
	a: string
}

#B: {
	_b: string
}

#X: #A | #B

l: [...#X]

l: [
	{_b: "bar"}
]
```

```
 cue export test.cue
l.0: cannot convert incomplete value "(C{a: string, _b: \"bar\"} | C{_b: (string & \"bar\")})" to JSON:
    ./test.cue:1:5
    ./test.cue:11:8
    ./test.cue:14:2
```

```
#B: {
	_b: string
}

l: [...#B]

l: [
	{_b: "bar"}
]
```

```
 cue export test.cue
{
    "l": [
        {}
    ]
}
```

**Comment by mpvl** [comment, marked as answer]
This is a bug. I can confirm this is fixed in the new evaluator. I'll convert this to an issue.

The core evaluator is now passing the vast majority of tests, but has a few small kinks to be ironed out. Getting closer and closer though!

**Comment by eonpatapon** [comment]
Great ! Thanks @mpvl

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/417.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
